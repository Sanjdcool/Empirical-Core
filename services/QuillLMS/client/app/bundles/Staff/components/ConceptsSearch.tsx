import React from 'react'
import {Icon, Input, AutoComplete} from 'antd'
import { Concept } from '../containers/ConceptsEdit';

const Option = AutoComplete.Option;
const OptGroup = AutoComplete.OptGroup;

export interface ConceptSearchProps {
  concepts: Array<Concept>
}

declare interface DataSourceChild {
  title: string
  id: string
}

declare interface DataSourceItem {
  title:string;
  children:Array<DataSourceChild>
}

function getGroupName(concept:Concept):string{
  return `${concept.parent && concept.parent.parent ? concept.parent.parent.name : 'NA'} > ${concept.parent ? concept.parent.name : 'NA'}`
}

function formatSearch(concepts:Array<Concept>):Array<DataSourceItem>{
  const groupings = {}
  concepts.forEach((concept) => {
    let group:Array<DataSourceChild>|null = groupings[getGroupName(concept)] 
    const newDataSourceChild:DataSourceChild = {
      title: concept.name,
      id: concept.id
    }
    group ? group.push(newDataSourceChild) : group = [newDataSourceChild];
  })

  const dataSource =  Object.keys(groupings).map((group) => {
    return {
      title: group,
      children: groupings[group]
    }
  });
  console.log(dataSource);
  return dataSource;
}

function renderOptions(concepts){
  console.log("rendering options")
  const dataSource = formatSearch(concepts)

  return dataSource.map(group => (
    <OptGroup
      key={group.title}
      label={group.title}
    >
      {group.children.map(opt => (
        <Option key={opt.id} value={opt.id}>
          {opt.title}
        </Option>
      ))}
    </OptGroup>
  ));
}

const ConceptSearch: React.SFC<ConceptSearchProps> =  ({concepts}) => {
  return (
    <div className="certain-category-search-wrapper" style={{ width: 250 }}>
      <AutoComplete
        className="certain-category-search"
        dropdownClassName="certain-category-search-dropdown"
        dropdownMatchSelectWidth={false}
        dropdownStyle={{ width: 300 }}
        size="large"
        style={{ width: '100%' }}
        dataSource={renderOptions(concepts)}
        placeholder="input here"
        optionLabelProp="value"
      >
        <Input suffix={<Icon type="search" className="certain-category-icon" />} />
      </AutoComplete>
    </div>
  )
}

export default ConceptSearch