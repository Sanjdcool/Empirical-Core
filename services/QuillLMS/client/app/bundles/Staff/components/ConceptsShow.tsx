import React from 'react';
import {Row, Col, Button} from 'antd';
import {Link} from 'react-router';
import ConceptsList from './ConceptsList';
import ConceptArchiveButton from './ConceptArchiveButton';
import {Concept} from '../containers/ConceptsIndex'

export interface ConceptShowProps {
  concept: Concept
}

const ConceptShow: React.SFC<ConceptShowProps> = ({concept}) => {
  return (
    <div>
      <h3>{concept.name}</h3>
      <p>{concept.uid}</p>
      <Button.Group>
        <Link to={`${concept.id}/edit`}>
        <Button type="default" icon="edit">Edit</Button>
        </Link>
        <ConceptArchiveButton id={concept.id} visible={concept.visible}/>
        <Button type="default" icon="fork">Find & Replace</Button>
      </Button.Group>
      <Row gutter={16}>
        <Col span={12}><ConceptsList concepts={concept.siblings} title={"Siblings"}/></Col>
        <Col span={12}><ConceptsList concepts={concept.children} title={"Children"}/></Col>
      </Row>
    </div>
  )
  
} 

export default ConceptShow