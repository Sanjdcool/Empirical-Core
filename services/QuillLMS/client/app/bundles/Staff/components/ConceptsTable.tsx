import * as React from 'react';
import { Table } from 'antd';
import { Concept } from '../containers/ConceptsIndex';


interface ConceptsTableProps {
  concepts: Array<Concept>
}

interface ConceptRow {
  key: string;
  conceptName:string;
  parentConceptName:string|null;
  grandparentConceptName:string|null;
  conceptId:string;
  parentConceptId:string|null;
  grandparentConceptId:string|null;
}

const columns = [
  {
    title: 'Grandparent Name',
    dataIndex: 'grandparentConceptName',
    key: 'grandparentConceptName',
    render: (text, record:ConceptRow) => (<a href={"/cms/concepts/" + record.grandparentConceptId}>{text}</a>),
  },
  {
    title: 'Parent Name',
    dataIndex: 'parentConceptName',
    key: 'parentConceptName',
    render: (text, record:ConceptRow) => (<a href={"/cms/concepts/" + record.parentConceptId}>{text}</a>),
  },
  {
    title: 'Name',
    dataIndex: 'conceptName',
    key: 'conceptName',
    render: (text, record:ConceptRow) => (<a href={"/cms/concepts/" + record.conceptId}>{text}</a>),
  },
];

function prepareData(data:Array<Concept>):Array<ConceptRow> {
  return data.map((concept:Concept) => prepareRow(concept));
}

function prepareRow(concept:Concept):ConceptRow {
  return {
    key: concept.id,
    conceptName: concept.name,
    conceptId: concept.id,
    parentConceptName: concept.parent ? concept.parent.name : null,
    parentConceptId: concept.parent ? concept.parent.id : null,
    grandparentConceptName: concept.parent && concept.parent.parent ? concept.parent.parent.name : null,
    grandparentConceptId: concept.parent && concept.parent.parent ? concept.parent.parent.id : null,
  }
}

const ConceptsTable: React.SFC<ConceptsTableProps> = ({concepts}) => {
  const data = prepareData(concepts);
  return (<Table columns={columns} dataSource={data} />);
};

export default ConceptsTable;