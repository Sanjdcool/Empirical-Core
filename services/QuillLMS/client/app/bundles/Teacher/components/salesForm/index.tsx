import * as React from 'react';

import LowerFormFields from './lowerFormFields';
import SchoolAndDistrictFields from './schoolAndDistrictFields';
import UpperFormFields from './upperFormFields';

import { getSchoolsAndDistricts, validateSalesForm, submitSalesForm } from '../../helpers/salesForms';
import { InputEvent } from '../../../Staff/interfaces/evidenceInterfaces';
import {
  FIRST_NAME, LAST_NAME, EMAIL, PHONE_NUMBER, ZIPCODE, SCHOOL_PREMIUM_ESTIMATE, TEACHER_PREMIUM_ESTIMATE,
  STUDENT_PREMIUM_ESTIMATE, COMMENTS, SCHOOL, DISTRICT, SCHOOL_OR_DISTRICT, SCHOOL_NOT_LISTED, DISTRICT_NOT_LISTED, SUBMISSION_ERROR
} from '../../../../constants/salesForm';

export const SalesForm = ({ type }) => {
  const [errors, setErrors] = React.useState<any>({});
  const [firstName, setFirstName] = React.useState<string>('');
  const [lastName, setLastName] = React.useState<string>('');
  const [email, setEmail] = React.useState<string>('');
  const [phoneNumber, setPhoneNumber] = React.useState<string>('');
  const [zipcode, setZipcode] = React.useState<string>('');
  const [schoolPremimumEstimate, setSchoolPremiumEstimate] = React.useState<string>('');
  const [teacherPremimumEstimate, setTeacherPremiumEstimate] = React.useState<string>('');
  const [studentPremimumEstimate, setStudentPremiumEstimate] = React.useState<string>('');
  const [comments, setComments] = React.useState<string>('');
  const [schools, setSchools] = React.useState<any[]>([]);
  const [districts, setDistricts] = React.useState<any[]>([]);
  const [selectedSchool, setSelectedSchool] = React.useState<string>('');
  const [schoolNotListed, setSchoolNotListed] = React.useState<boolean>(false);
  const [selectedDistrict, setSelectedDistrict] = React.useState<string>('');
  const [districtNotListed, setDistrictNotListed] = React.useState<boolean>(false);
  const [schoolOrDistrict, setSchoolOrDistrict] = React.useState<any>('');
  const [formSubmitted, setFormSubmitted] = React.useState<boolean>(false);

  React.useEffect(() => {
    if(!schools.length) {
      getSchoolsAndDistricts('school').then((response) => {
        if(response && response.options) {
          setSchools(response.options)
        }
      });
    }
    if(!districts.length) {
      getSchoolsAndDistricts('district').then((response) => {
        if(response && response.options) {
          setDistricts(response.options)
        }
      });
    }
  }, []);

  const stateSetters = {
    [FIRST_NAME]: setFirstName,
    [LAST_NAME]: setLastName,
    [EMAIL]: setEmail,
    [PHONE_NUMBER]: setPhoneNumber,
    [ZIPCODE]: setZipcode,
    [SCHOOL_PREMIUM_ESTIMATE]: setSchoolPremiumEstimate,
    [TEACHER_PREMIUM_ESTIMATE]: setTeacherPremiumEstimate,
    [STUDENT_PREMIUM_ESTIMATE]: setStudentPremiumEstimate,
    [COMMENTS]: setComments,
    [SCHOOL]: setSelectedSchool,
    [DISTRICT]: setSelectedDistrict,
    [SCHOOL_OR_DISTRICT]: setSchoolOrDistrict
  }
  const schoolIsSelected = schoolOrDistrict === SCHOOL;
  const districtIsSelected = schoolOrDistrict === DISTRICT;

  function handleUpdateField(e: InputEvent | React.ChangeEvent<HTMLTextAreaElement>) {
    const { target } = e;
    const { value, id } = target;
    const setterFunction = stateSetters[id];
    if(id === SCHOOL_OR_DISTRICT) {
      // reset to displaying search inputs in case user changes from school to district option (and vice versa)
      setSchoolNotListed(false);
      setDistrictNotListed(false);
    }
    setterFunction(value);
  }

  function handleSchoolSearchChange(value: string) {
    if(value === SCHOOL_NOT_LISTED) {
      setSchoolNotListed(true);
      setSelectedSchool('');
    } else {
      setSelectedSchool(value);
    }
  };

  function handleDistrictSearchChange(value: string) {
    if(value === DISTRICT_NOT_LISTED) {
      setDistrictNotListed(true);
      setSelectedDistrict('');
    } else {
      setSelectedDistrict(value);
    }
  };

  function handleFormSubmission(e: React.MouseEvent) {
    e.preventDefault();
    const salesFormSubmission = {
      first_name: firstName,
      last_name: lastName,
      email: email,
      phone_number: phoneNumber,
      zipcode: zipcode,
      collection_type: schoolOrDistrict.toLowerCase(),
      school_name: selectedSchool,
      district_name: selectedDistrict,
      school_premium_count_estimate: parseInt(schoolPremimumEstimate),
      teacher_premium_count_estimate: parseInt(teacherPremimumEstimate),
      student_premium_count_estimate: parseInt(studentPremimumEstimate),
      submission_type: type,
      comment: comments
    }
    const formErrors = validateSalesForm(salesFormSubmission);
    if(Object.keys(formErrors).length) {
      setErrors(formErrors)
    } else {
      setErrors({});
      submitSalesForm(salesFormSubmission).then(response => {
        if(response.error) {
          const submissionError = { [SUBMISSION_ERROR]: response.error };
          setErrors(submissionError)
        } else {
          setFormSubmitted(true);
        }
      });
    }
  }

  return(
    <div className="sales-form-container">
      {!formSubmitted && <form className="container">
        <UpperFormFields
          email={email}
          errors={errors}
          firstName={firstName}
          handleUpdateField={handleUpdateField}
          lastName={lastName}
          phoneNumber={phoneNumber}
          type={type}
          zipcode={zipcode}
        />
        <SchoolAndDistrictFields
          districtIsSelected={districtIsSelected}
          districtNotListed={districtNotListed}
          districts={districts}
          errors={errors}
          handleDistrictSearchChange={handleDistrictSearchChange}
          handleSchoolSearchChange={handleSchoolSearchChange}
          handleUpdateField={handleUpdateField}
          schoolIsSelected={schoolIsSelected}
          schoolNotListed={schoolNotListed}
          schools={schools}
          selectedDistrict={selectedDistrict}
          selectedSchool={selectedSchool}
        />
        <LowerFormFields
          comments={comments}
          errors={errors}
          handleUpdateField={handleUpdateField}
          schoolPremimumEstimate={schoolPremimumEstimate}
          studentPremimumEstimate={studentPremimumEstimate}
          teacherPremimumEstimate={teacherPremimumEstimate}
        />
        {errors[SUBMISSION_ERROR] && <p className="error-text">{errors[SUBMISSION_ERROR]}</p>}
        <button className="submit-button quill-button contained primary medium" onClick={handleFormSubmission}>Submit</button>
      </form>}
      {formSubmitted && <div className="success-container"><h3>Request successfully submitted!</h3></div>}
    </div>
  )
}

export default SalesForm
