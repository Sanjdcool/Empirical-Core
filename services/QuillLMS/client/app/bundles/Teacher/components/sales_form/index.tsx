import * as React from 'react';
import SelectSearch from 'react-select-search';

import { titleCase, Input } from '../../../Shared';
import { InputEvent } from '../../../Staff/interfaces/evidenceInterfaces';
import { getSchoolsAndDistricts, schoolSearch, districtSearch, SCHOOL_NOT_LISTED, DISTRICT_NOT_LISTED } from '../../helpers/salesForms';

const FIRST_NAME = 'First name';
const LAST_NAME = 'Last name';
const EMAIL = 'Email';
const PHONE_NUMBER = 'Phone number';
const ZIPCODE = 'Zipcode';
const SCHOOL_PREMIUM_ESTIMATE = 'Estimated number of schools that will receive Quill Premium';
const TEACHER_PREMIUM_ESTIMATE = 'Estimated number of teachers that will receive Quill Premium';
const STUDENT_PREMIUM_ESTIMATE = 'Estimated number of students that will receive Quill Premium';
const COMMENTS = 'Comments';
const SCHOOL = 'School';
const DISTRICT = 'District';
const SCHOOL_OR_DISTRICT = 'School or district';

export const SalesForm = ({ type }) => {
  const [errors, setErrors] = React.useState<string>('');
  const [firstName, setFirstName] = React.useState<string>('');
  const [lastName, setLastName] = React.useState<string>('');
  const [email, setEmail] = React.useState<string>('');
  const [phoneNumber, setPhoneNumber] = React.useState<string>('');
  const [zipcode, setZipcode] = React.useState<string>('');
  const [schoolPremimumEstimate, setSchoolPremiumEstimate] = React.useState<string>('');
  const [teacherPremimumEstimate, setTeacherPremiumEstimate] = React.useState<string>('');
  const [studentPremimumEstimate, setStudentPremiumEstimate] = React.useState<string>('');
  const [comments, setComments] = React.useState<string>('');
  const [schools, setSchools] = React.useState<any[]>([]);
  const [districts, setDistricts] = React.useState<any[]>([]);
  const [selectedSchool, setSelectedSchool] = React.useState<string>('');
  const [schoolNotListed, setSchoolNotListed] = React.useState<boolean>(false);
  const [selectedDistrict, setSelectedDistrict] = React.useState<string>('');
  const [districtNotListed, setDistrictNotListed] = React.useState<boolean>(false);
  const [schoolOrDistrict, setSchoolOrDistrict] = React.useState<any>('');

  React.useEffect(() => {
    if(!schools.length) {
      getSchoolsAndDistricts('school').then((response) => {
        if(response && response.options) {
          setSchools(response.options)
        }
      });
    }
    if(!districts.length) {
      getSchoolsAndDistricts('district').then((response) => {
        if(response && response.options) {
          setDistricts(response.options)
        }
      });
    }
  }, []);

  const stateSetters = {
    [FIRST_NAME]: setFirstName,
    [LAST_NAME]: setLastName,
    [EMAIL]: setEmail,
    [PHONE_NUMBER]: setPhoneNumber,
    [ZIPCODE]: setZipcode,
    [SCHOOL_PREMIUM_ESTIMATE]: setSchoolPremiumEstimate,
    [TEACHER_PREMIUM_ESTIMATE]: setTeacherPremiumEstimate,
    [STUDENT_PREMIUM_ESTIMATE]: setStudentPremiumEstimate,
    [COMMENTS]: setComments,
    [SCHOOL]: setSelectedSchool,
    [DISTRICT]: setSelectedDistrict,
    [SCHOOL_OR_DISTRICT]: setSchoolOrDistrict
  }
  const schoolIsSelected = schoolOrDistrict === SCHOOL;
  const districtIsSelected = schoolOrDistrict === DISTRICT;

  function handleUpdateField(e: InputEvent | React.ChangeEvent<HTMLTextAreaElement>) {
    const { target } = e;
    const { value, id } = target;
    const setterFunction = stateSetters[id];
    setterFunction(value);
  }

  function handleSchoolSearchChange(value) {
    if(value === SCHOOL_NOT_LISTED) {
      setSchoolNotListed(true);
      setSelectedSchool('');
    } else {
      setSelectedSchool(value);
    }
  };

  function handleDistrictSearchChange(value) {
    if(value === DISTRICT_NOT_LISTED) {
      setDistrictNotListed(true);
      setSelectedDistrict('');
    } else {
      setSelectedDistrict(value);
    }
  };

  function renderSchoolOrDistrictSelect() {
    if(schoolIsSelected && !schoolNotListed) {
      return(
        <div>
          <SelectSearch
            filterOptions={schoolSearch}
            onChange={handleSchoolSearchChange}
            options={schools}
            placeholder="Search for your school"
            search={true}
          />
        </div>
      );
    }
    if(schoolNotListed) {
      return(
        <Input
          className="school"
          handleChange={handleUpdateField}
          id={SCHOOL}
          label={SCHOOL}
          placeholder=""
          value={selectedSchool}
        />
      );
    }
    if(districtIsSelected && !districtNotListed) {
      return(
        <div>
          <SelectSearch
            filterOptions={districtSearch}
            onChange={handleDistrictSearchChange}
            options={districts}
            placeholder="Search for your district"
            search={true}
          />
        </div>
      );
    }
    if(districtNotListed) {
      return(
        <Input
          className="district"
          handleChange={handleUpdateField}
          id={DISTRICT}
          label={DISTRICT}
          placeholder=""
          value={selectedDistrict}
        />
      );
    }
  }
  return(
    <div className="container">
      <h3>{`${titleCase(type)} form`}</h3>
      <form>
        <Input
          className="first-name"
          error={errors[FIRST_NAME]}
          handleChange={handleUpdateField}
          id={FIRST_NAME}
          label={FIRST_NAME}
          placeholder=""
          value={firstName}
        />
        <Input
          className="last-name"
          error={errors[FIRST_NAME]}
          handleChange={handleUpdateField}
          id={LAST_NAME}
          label={LAST_NAME}
          placeholder=""
          value={lastName}
        />
        <Input
          className="email"
          error={errors[EMAIL]}
          handleChange={handleUpdateField}
          id={EMAIL}
          label={EMAIL}
          placeholder=""
          type="email"
          value={email}
        />
        <Input
          characterLimit={10}
          className="phone-number"
          error={errors[PHONE_NUMBER]}
          handleChange={handleUpdateField}
          id={PHONE_NUMBER}
          label={PHONE_NUMBER}
          placeholder=""
          type="tel"
          value={phoneNumber}
        />
        <Input
          characterLimit={5}
          className="zipcode"
          error={errors[ZIPCODE]}
          handleChange={handleUpdateField}
          id={ZIPCODE}
          label={ZIPCODE}
          placeholder=""
          value={zipcode}
        />
        <div>
          <p>Do you represent a school or district?</p>
          <div className="radio-options">
            <div className="radio">
              <label htmlFor={SCHOOL}>
                <input checked={schoolIsSelected} id={SCHOOL_OR_DISTRICT} onChange={handleUpdateField} type="radio" value={SCHOOL} />
                {SCHOOL}
              </label>
            </div>
            <div className="radio">
              <label htmlFor={DISTRICT}>
                <input checked={districtIsSelected} id={SCHOOL_OR_DISTRICT} onChange={handleUpdateField} type="radio" value={DISTRICT} />
                {DISTRICT}
              </label>
            </div>
          </div>
        </div>
        {renderSchoolOrDistrictSelect()}
        <Input
          className="school-premium-estimate"
          error={errors[SCHOOL_PREMIUM_ESTIMATE]}
          handleChange={handleUpdateField}
          id={SCHOOL_PREMIUM_ESTIMATE}
          label={SCHOOL_PREMIUM_ESTIMATE}
          placeholder=""
          value={schoolPremimumEstimate}
        />
        <Input
          className="teacher-premium-estimate"
          error={errors[TEACHER_PREMIUM_ESTIMATE]}
          handleChange={handleUpdateField}
          id={TEACHER_PREMIUM_ESTIMATE}
          label={TEACHER_PREMIUM_ESTIMATE}
          placeholder=""
          value={teacherPremimumEstimate}
        />
        <Input
          className="student-premium-estimate"
          error={errors[STUDENT_PREMIUM_ESTIMATE]}
          handleChange={handleUpdateField}
          id={STUDENT_PREMIUM_ESTIMATE}
          label={STUDENT_PREMIUM_ESTIMATE}
          placeholder=""
          value={studentPremimumEstimate}
        />
        <div className="control">
          <label className="label" htmlFor={COMMENTS} id="comments-label">Comments (optional)</label>
          <textarea
            aria-labelledby="comments-label"
            id={COMMENTS}
            onChange={handleUpdateField}
            style={{minHeight: '100px', border: '1px solid black', padding: '10px', width: '100%'}}
            value={comments}
          />
        </div>
      </form>
    </div>
  )
}

export default SalesForm
