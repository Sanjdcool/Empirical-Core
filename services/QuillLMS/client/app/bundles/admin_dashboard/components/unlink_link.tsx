import { truncate, truncateSync } from 'fs';
import * as React from 'react';
import { requestPost } from '../../../modules/request/index.js';

const unlink = (id) => {
  return requestPost(`/teachers/unlink/${id}`, null, () => {
    alert("unlinked")
  })
}

const UnlinkLink = ({ id }) => {
  return <a onClick={() => unlink(id)}>Unlink From School</a>
}

export default UnlinkLink;
