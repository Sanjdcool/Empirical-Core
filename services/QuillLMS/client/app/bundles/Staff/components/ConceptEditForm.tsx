import * as React from "react";
import {Link} from "react-router";
import { Query, Mutation } from "react-apollo";
import gql from "graphql-tag";
import { Concept } from "../containers/ConceptsShow";
import ConceptForm from './ConceptForm';

const EDIT_CONCEPT = gql`
mutation editConcept($id: ID! $name: String, $parentId: ID){
    editConcept(input: {id: $id, name: $name, parentId: $parentId}){
      concept {
        id
        uid
        name
        parentId
      }
    }
  }
`;


export interface AppProps {
  concept: Concept
  redirectToShow(Concept): null
}

function getParentIdArray(concept:Concept):Array<Number>{
  const parentIdArray = [];
  const grandparentId:Number = concept.parent && concept.parent.parent ? parentIdArray.push(concept.parent.parent.id) : null;
  const parentId:Number = concept.parent ? parentIdArray.push(concept.parent.id) : null;
  return parentIdArray;
} 

class ConceptEditForm extends React.Component<AppProps, any> {
  constructor(props){
    super(props)
    this.state = {
      fields: {
        name: {
          value: props.concept.name,
        },
        parentId: {
          value: getParentIdArray(props.concept),
        },
      },
    };
  }

  handleFormChange = (changedFields) => {
    this.setState(({ fields }) => {
      const newState =  {
        fields: { ...fields, ...changedFields },
      }
      return newState;
    });
  }

  handleFormSubmit = (e) => {
    e.preventDefault()
  }

  redirectToShow = (data) => {
    
    this.props.redirectToShow(data.editConcept.concept)
  }

  render() {
    const fields = this.state.fields;
    return (
      <Mutation mutation={EDIT_CONCEPT} onCompleted={this.redirectToShow}>
        {(editConcept, { data }) => (
          <ConceptForm {...fields} onChange={this.handleFormChange} formSubmitCopy={"Edit"} onSubmit={(e) => {
            e.preventDefault();
            editConcept({ variables: {id: this.props.concept.id, name: this.state.fields.name.value, parentId: this.state.fields.parentId.value[this.state.fields.parentId.value.length - 1]}});
          }} />
        )}
      </Mutation>
    )
  }
};

export default ConceptEditForm