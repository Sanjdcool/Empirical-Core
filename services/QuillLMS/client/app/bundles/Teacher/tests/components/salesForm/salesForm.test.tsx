import * as React from 'react';
import { mount } from 'enzyme';

import { SalesForm }from '../../../components/salesForm';

jest.mock('fuse.js', () => ({
  default: jest.fn()
}));

describe('SalesForm Component', () => {
  const component = mount(<SalesForm type="renewal request" />);

  it('should match snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
