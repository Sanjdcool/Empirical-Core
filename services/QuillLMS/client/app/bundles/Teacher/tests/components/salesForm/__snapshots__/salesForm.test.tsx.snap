// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SalesForm Component should match snapshot 1`] = `
<SalesForm
  type="renewal request"
>
  <div
    className="sales-form-container"
  >
    <form
      className="container"
    >
      <UpperFormFields
        email=""
        errors={Object {}}
        firstName=""
        handleUpdateField={[Function]}
        lastName=""
        phoneNumber=""
        type="renewal request"
        zipcode=""
      >
        <h3
          className="form-header"
        >
          Renewal request form
        </h3>
        <p
          className="disclaimer"
        >
          Please fill out this form. We will respond with a quote within three business days.
        </p>
        <Input
          autoComplete="on"
          className="form-input first-name"
          handleChange={[Function]}
          id="First name"
          label="First name"
          placeholder=""
          value=""
        >
          <div
            className="input-container  inactive  form-input first-name "
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={-1}
          >
            <label
              htmlFor="First name"
            >
              First name
            </label>
            <input
              autoComplete="on"
              id="First name"
              maxLength={10000}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder={false}
              value=""
            />
          </div>
        </Input>
        <Input
          autoComplete="on"
          className="form-input last-name"
          handleChange={[Function]}
          id="Last name"
          label="Last name"
          placeholder=""
          value=""
        >
          <div
            className="input-container  inactive  form-input last-name "
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={-1}
          >
            <label
              htmlFor="Last name"
            >
              Last name
            </label>
            <input
              autoComplete="on"
              id="Last name"
              maxLength={10000}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder={false}
              value=""
            />
          </div>
        </Input>
        <Input
          autoComplete="on"
          className="form-input email"
          handleChange={[Function]}
          id="Email"
          label="Email"
          placeholder=""
          type="email"
          value=""
        >
          <div
            className="input-container  inactive  form-input email "
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={-1}
          >
            <label
              htmlFor="Email"
            >
              Email
            </label>
            <input
              autoComplete="on"
              id="Email"
              maxLength={10000}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder={false}
              type="email"
              value=""
            />
          </div>
        </Input>
        <Input
          autoComplete="on"
          characterLimit={10}
          className="form-input phone"
          handleChange={[Function]}
          id="Phone number"
          label="Phone number"
          placeholder=""
          type="tel"
          value=""
        >
          <div
            className="input-container  inactive  form-input phone has-character-limit"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={-1}
          >
            <label
              htmlFor="Phone number"
            >
              Phone number
            </label>
            <input
              autoComplete="on"
              id="Phone number"
              maxLength={10}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder={false}
              type="tel"
              value=""
            />
            <div
              className="character-limit"
            >
              <span>
                0
                /
                10
              </span>
            </div>
          </div>
        </Input>
        <Input
          autoComplete="on"
          characterLimit={5}
          className="form-input zipcode"
          handleChange={[Function]}
          id="Zipcode"
          label="Zipcode"
          placeholder=""
          value=""
        >
          <div
            className="input-container  inactive  form-input zipcode has-character-limit"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={-1}
          >
            <label
              htmlFor="Zipcode"
            >
              Zipcode
            </label>
            <input
              autoComplete="on"
              id="Zipcode"
              maxLength={5}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder={false}
              value=""
            />
            <div
              className="character-limit"
            >
              <span>
                0
                /
                5
              </span>
            </div>
          </div>
        </Input>
      </UpperFormFields>
      <SchoolAndDistrictFields
        districtIsSelected={false}
        districtNotListed={false}
        districts={Array []}
        errors={Object {}}
        handleDistrictSearchChange={[Function]}
        handleSchoolSearchChange={[Function]}
        handleUpdateField={[Function]}
        schoolIsSelected={false}
        schoolNotListed={false}
        schools={Array []}
        selectedDistrict=""
        selectedSchool=""
      >
        <div
          className="school-or-district-container"
        >
          <p>
            Do you represent a school or district?
          </p>
          <div
            className="radio-options"
          >
            <div
              className="radio"
            >
              <label
                htmlFor="School"
              >
                <input
                  checked={false}
                  id="School or district"
                  onChange={[Function]}
                  type="radio"
                  value="School"
                />
                School
              </label>
            </div>
            <div
              className="radio"
            >
              <label
                htmlFor="District"
              >
                <input
                  checked={false}
                  id="School or district"
                  onChange={[Function]}
                  type="radio"
                  value="District"
                />
                District
              </label>
            </div>
          </div>
        </div>
        <div>
          <ForwardRef
            filterOptions={[Function]}
            onChange={[Function]}
            options={Array []}
            placeholder="Search for your school"
            search={true}
          >
            <div
              className="select-search"
              id={null}
            >
              <div
                className="select-search__value"
              >
                <input
                  autoComplete="on"
                  autoFocus={false}
                  className="select-search__input"
                  disabled={false}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyPress={[Function]}
                  onKeyUp={[Function]}
                  placeholder="Search for your school"
                  readOnly={false}
                  tabIndex="0"
                  value=""
                />
              </div>
            </div>
          </ForwardRef>
          <ForwardRef
            filterOptions={[Function]}
            onChange={[Function]}
            options={Array []}
            placeholder="Search for your district"
            search={true}
          >
            <div
              className="select-search"
              id={null}
            >
              <div
                className="select-search__value"
              >
                <input
                  autoComplete="on"
                  autoFocus={false}
                  className="select-search__input"
                  disabled={false}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyPress={[Function]}
                  onKeyUp={[Function]}
                  placeholder="Search for your district"
                  readOnly={false}
                  tabIndex="0"
                  value=""
                />
              </div>
            </div>
          </ForwardRef>
        </div>
      </SchoolAndDistrictFields>
      <LowerFormFields
        comments=""
        errors={Object {}}
        handleUpdateField={[Function]}
        schoolPremimumEstimate=""
        studentPremimumEstimate=""
        teacherPremimumEstimate=""
      >
        <Input
          autoComplete="on"
          className="form-input estimate"
          handleChange={[Function]}
          id="Estimated number of schools that will receive Quill Premium"
          label="Estimated number of schools that will receive Quill Premium"
          placeholder=""
          value=""
        >
          <div
            className="input-container  inactive  form-input estimate "
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={-1}
          >
            <label
              htmlFor="Estimated number of schools that will receive Quill Premium"
            >
              Estimated number of schools that will receive Quill Premium
            </label>
            <input
              autoComplete="on"
              id="Estimated number of schools that will receive Quill Premium"
              maxLength={10000}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder={false}
              value=""
            />
          </div>
        </Input>
        <Input
          autoComplete="on"
          className="form-input estimate"
          handleChange={[Function]}
          id="Estimated number of teachers that will receive Quill Premium"
          label="Estimated number of teachers that will receive Quill Premium"
          placeholder=""
          value=""
        >
          <div
            className="input-container  inactive  form-input estimate "
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={-1}
          >
            <label
              htmlFor="Estimated number of teachers that will receive Quill Premium"
            >
              Estimated number of teachers that will receive Quill Premium
            </label>
            <input
              autoComplete="on"
              id="Estimated number of teachers that will receive Quill Premium"
              maxLength={10000}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder={false}
              value=""
            />
          </div>
        </Input>
        <Input
          autoComplete="on"
          className="form-input estimate"
          handleChange={[Function]}
          id="Estimated number of students that will receive Quill Premium"
          label="Estimated number of students that will receive Quill Premium"
          placeholder=""
          value=""
        >
          <div
            className="input-container  inactive  form-input estimate "
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={-1}
          >
            <label
              htmlFor="Estimated number of students that will receive Quill Premium"
            >
              Estimated number of students that will receive Quill Premium
            </label>
            <input
              autoComplete="on"
              id="Estimated number of students that will receive Quill Premium"
              maxLength={10000}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder={false}
              value=""
            />
          </div>
        </Input>
        <div
          className="comments-container"
        >
          <label
            htmlFor="Comments"
            id="comments-label"
          >
            Comments (optional)
          </label>
          <textarea
            aria-labelledby="comments-label"
            id="Comments"
            onChange={[Function]}
            style={
              Object {
                "border": "1px solid black",
                "minHeight": "100px",
                "padding": "10px",
                "width": "100%",
              }
            }
            value=""
          />
        </div>
      </LowerFormFields>
      <button
        className="submit-button quill-button contained primary medium"
        onClick={[Function]}
      >
        Submit
      </button>
    </form>
  </div>
</SalesForm>
`;
